api/test/robot_tests/plugin_manager.robot

```diff
@@ -2,20 +2,215 @@
+
+POST /plugin-manager/{id}/execute with invalid id
+    ${data}=    Create Dictionary    command=start
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/invalid/execute
+    ...    expected_status=${STATUS_NOT_FOUND}
+    ...    json=${data}
+    Should Be Equal As Strings    ${response.json()['code']}    Plugin-0001
+    Should Be Equal As Strings    ${response.json()['message']}    get plugin error: no plugin matching id invalid
+
+DELETE /plugin-manager/{id} with invalid id
+    ${response}=    DELETE    ${API_URL}/plugin-manager/3829029    expected_status=${STATUS_INTERNAL_SERVER_ERROR}
+    Should Be Equal As Strings
+    ...    ${response.json()['message']}
+    ...    deleting plugin 3829029: no plugin matching id 3829029
+
+POST /plugin-manager/{id}/execute with invalid body
+    ${data}=    Create Dictionary    command=test
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/invalid/execute
+    ...    expected_status=${STATUS_UNPROCESSABLE_ENTITY}
+    ...    json=${data}
+    Should Be Equal As Strings    ${response.json()['code']}    606
+    Should Be Equal As Strings
+    ...    ${response.json()['message']}
+    ...    body.command in body should be one of [update stop start restart]
+
+POST /plugin-manager/register with invalid id
+    ${data}=    Create Dictionary
+    ...    id=1
+    ...    name=ut aliqua non
+    ...    author=adipisicing
+    ...    description=minim consectetur dolore,
+    ...    logo=id et sunt irure,
+    ...    home=sunt
+    ...    api_spec=culpa enim sint aliqua
+    ...    url=oluptate
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/register
+    ...    expected_status=${STATUS_NOT_FOUND}
+    ...    json=${data}
+
+    Should Be Equal As Strings    ${response.json()['code']}    Plugin-0001
+    Should Be Equal As Strings    ${response.json()['message']}    get plugin error: no plugin matching id 1
+
+POST /plugin-manager/register with invalid body
+    ${data}=    Create Dictionary
+    ...    id=-65217
+    ...    name=ut aliqua non
+    ...    author=adipisicing
+    ...    description=minim consectetur dolore,
+    ...    logo=id et sunt irure,
+    ...    home=sunt
+    ...    api_spec=culpa enim sint aliqua
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/register
+    ...    expected_status=${STATUS_UNPROCESSABLE_ENTITY}
+    ...    json=${data}
+    Should Be Equal As Strings    ${response.json()['code']}    602
+    Should Be Equal As Strings    ${response.json()['message']}    body.url in body is required
+
+POST /plugin-manager/{id}/execute with NotImplemented update command
+    ${id}=    Get Plugin ID From Author and Name    massalabs    hello world
+    ${data}=    Create Dictionary    command=update
+    ${response}=    POST
+    ...    ${API_URL}/plugin-manager/${id}/execute
+    ...    expected_status=${STATUS_NOT_IMPLEMENTED}
+    ...    json=${data}
+
+
+*** Keywords ***
```

Comment:
@gregLibert I decided to put test specific to plugins inside this file since we shouldn't need them outside of this file.

Another way of doing it would be to create a sub directory for each part we do, and so, have the keywords specific to a part in a specific `keywords.resource` file in that subdirectory. Same for variables. We would end up with something like that:
```
- robot_tests/
   - plugins/
      - plugins.robot
      - variables.resource
      - keywords.resource
    - cmd/
      - cmd.robot
      - variables.resource
...
```

I didn't got that way because we are not sure that every parts of Thyra will need some specific variables and keywords. 

WDYT ?